pkg/cosign/verify.go

```diff
@@ -291,6 +291,88 @@ func ValidateAndUnpackCert(cert *x509.Certificate, co *CheckOpts) (signature.Ver
 	return verifier, nil
 }
 
+// ValidateAndUnpackCertWithIntermediates function has same logic as ValidateAndUnpackCert.
+// Main diffrence is that we can not use ValidateAndUnpackCert in multi thread invocation
+// as it uses a common reference of CheckOpts so in multi thread invocation co.intermediateCert object
+// can get overwritten my multiple threads. So to solve this ValidateAndUnpackCertWithIntermediates
+// is created which has a separate argument for Intermediate certs.
+func ValidateAndUnpackCertWithIntermediates(cert *x509.Certificate, co *CheckOpts, intermediateCerts *x509.CertPool) (signature.Verifier, error) {
```

Comment:
We should also write a unit test for this. It just needs one, to demonstrate it uses the passed-in intermediates rather than the ones from `co` like `ValidateAndUnpackCerts`

Reply:
Agreed i have add a new Unit test case TestValidateAndUnpackCertWithIntermediatesSuccess() in this i dont pass intermediatecert in co.IntermediateCert but just pass it in the argument to the function ValidateAndUnpackCertWithIntermediates() so it pick the value from argument